### PR TITLE
CSS: scroll snap module

### LINKED ISSUE #155: web/css/css_scroll_snap/ -
https://github.com/mdn/content/pull/26735/

### PR DESCRIPTION
**Pull request by estelle**
Part of https://github.com/openwebdocs/project/issues/147

* brief description
* demonstration example
* properties
* guides
* associated properties
* glossary terms
Fixes https://github.com/openwebdocs/project/issues/155

### CODE REVIEW

**Review by dipikabh — CHANGES_REQUESTED** [review]
Thanks @estelle for making these updates and adding "the example"!

I've added some edit suggestions.

File: files/en-us/web/css/css_scroll_snap/basic_concepts/index.md [hunk 1]
```diff
@@ -6,16 +6,20 @@ page-type: guide
...
 
-The example below demonstrates scroll snapping along the vertical axis, which is defined by `scroll-snap-type`. Additionally, `scroll-snap-align` is used on all the `<section>` element children dictating the point where the scrolling of each child should stop.
+You can then define scroll snapping on the scroll container by using the following two key properties:
+- {{cssxref("scroll-snap-type")}}: Using this property, you can define whether or not the scrollable viewport can be snapped to, whether snapping is required or optional, and the axis on which the snapping should occur.
+- {{cssxref("scroll-snap-align")}}: This property is set on every child of the scroll container and you can use it to define each child's snap position or lack thereof. (Using the {{cssxref("scroll-snap-stop")}} property, you can ensure that a child is snapped to during scrolling and not passed over. You can also set several {{cssxref("scroll-margin")}} properties on child elements that are snapped to during scrolling to create an outset from the defined box.)
+
+
```
Comment (estelle) [inline files/en-us/web/css/css_scroll_snap/basic_concepts/index.md:21]:
```suggestion

```

File: files/en-us/web/css/css_scroll_snap/basic_concepts/index.md [hunk 2]
```diff
@@ -6,14 +6,22 @@ page-type: guide
 
 {{CSSRef}}
 
-The [CSS Scroll Snap](/en-US/docs/Web/CSS/CSS_Scroll_Snap) feature provides a way to snap the scrolling to certain points as the user scrolls through a document. This can be helpful in creating a more app-like experience on mobile or even on the desktop for some types of applications.
+The [CSS Scroll Snap](/en-US/docs/Web/CSS/CSS_Scroll_Snap) feature provides a way to snap the scrolling to certain points as the user scrolls through a document. This can help create a more app-like experience on mobile or even on the desktop for some types of applications.
+
+Scroll snap defines scroll snap positions, which enforce the scroll positions that a scroll container's {{Glossary("scrollport")}} may end at, or "snap to," after a scrolling operation has completed.
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/css_scroll_snap/basic_concepts/index.md]: > * Not sure what "app-like experience" is referring to

It was there so i left it there.

File: files/en-us/web/css/css_scroll_snap/basic_concepts/index.md [hunk 3]
```diff
@@ -6,14 +6,22 @@ page-type: guide
...
 The key properties for the scroll snap feature are:
 
-- {{CSSxRef("scroll-snap-type")}}: This property is used on the [scroll container](/en-US/docs/Glossary/Scroll_container) to specify the type, direction, and optionality of scrolling.
-- {{CSSxRef("scroll-snap-align")}}: This property is used on child elements to specify the scroll snap position.
+- {{CSSxRef("scroll-snap-type")}}
+  - : Set on the [scroll container](/en-US/docs/Glossary/Scroll_container) to specify the type, direction, and optionality of scrolling.
+- {{CSSxRef("scroll-snap-align")}}: This property is used
+  - : Set on child elements to specify the scroll snap position.
 
 The example below demonstrates scroll snapping along the vertical axis, which is defined by `scroll-snap-type`. Additionally, `scroll-snap-align` is used on all the `<section>` element children dictating the point where the scrolling of each child should stop.
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/css_scroll_snap/basic_concepts/index.md]: stop is correct

File: files/en-us/web/css/css_scroll_snap/basic_concepts/index.md [hunk 4]
```diff
@@ -6,16 +6,18 @@ page-type: guide
...
-- {{CSSxRef("scroll-snap-align")}}: This property is used on child elements to specify the scroll snap position.
+Before you can define scroll snapping, you need to enable scrolling on a scroll container. You can do this by ensuring that the scroll container has a defined size and that it has {{cssxref("overflow")}} enabled. (Optional {{cssxref("scroll-padding")}} properties can be set on the scroll container to create a snapping offset.)
 
-The example below demonstrates scroll snapping along the vertical axis, which is defined by `scroll-snap-type`. Additionally, `scroll-snap-align` is used on all the `<section>` element children dictating the point where the scrolling of each child should stop.
+You can then define scroll snapping on the scroll container by using the following two key properties:
+- {{cssxref("scroll-snap-type")}}: Using this property, you can define whether or not the scrollable viewport can be snapped to, whether snapping is required or optional, and the axis on which the snapping should occur.
```
Comment (estelle) [inline files/en-us/web/css/css_scroll_snap/basic_concepts/index.md:18]:
```suggestion

- {{cssxref("scroll-snap-type")}}: Using this property, you can define whether or not the scrollable viewport can be snapped to, whether snapping is required or optional, and the axis on which the snapping should occur.
```

File: files/en-us/web/css/css_scroll_snap/basic_concepts/index.md [hunk 5]
```diff
@@ -6,14 +6,22 @@ page-type: guide
...
 
 ## Key properties for CSS Scroll Snap
 
+To enable scroll snapping, the scrolling behavior is defined on the scroll container. The container's {{cssxref("scroll-snap-type")}} property defines whether the scrollable viewport can be snapped to, the axis upon which the snapping occurs, and whether snapping is required or optional. To enable scrolling, the container should have a defined size and {{cssxref("overflow")}} must be enabled. There are optional {{cssxref("scroll-padding")}} properties that can be set on the scroll container to create a snapping offset.
+
+The {{cssxref("scroll-snap-align")}} property is set on every child of the scroll container, defining each child's snap position or lack thereof. The {{cssxref("scroll-snap-stop")}} property enables requiring that child is snapped to during scrolling and not passed over. Several {{cssxref("scroll-margin")}} properties can be set on the snapped-to-child elements to create an outset from the defined box.
+
 The key properties for the scroll snap feature are:
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/css_scroll_snap/basic_concepts/index.md]: > * I prefer the previous bulleted list style to list the related properties here. The bulleted list also makes the presentation on this page slightly different from the usual definition list on a property reference page.

This is the guide, not the module landing page. Does this comment still hold true?

[the remark this answers is not in the document]
Comment (dipikabh) [reply files/en-us/web/css/css_scroll_snap/basic_concepts/index.md]: Yes :)

**Review by dipikabh — APPROVED** [review]
Just one fix needed. Otherwise this looks good. Thank you 👍

**Pull request merged**